build_projects/shared-build-targets-utils/Utils/HostVersion.cs

```diff
@@ -107,6 +107,8 @@ public override string ToString()
         //
         public bool IsLocked = true; // Set this variable to toggle muxer locking.
         public VerInfo LockedHostFxrVersion => IsLocked ? new VerInfo(1, 0, 1, "", "", "", CommitCountString) : LatestHostFxrVersion;
+        //We add a different LockedHostFxrVersion for the MSI as the package needs to deviate for issue #1574. This we can update or remove when we have a binary change for the hostfxr.dll (from 1.0.1)
```

Comment:
New changes have been submitted based on the feedback. 

To summarize: 
 - All the 4 scenarios are now covered.
 - We have added a new variable LockedHostFxrMSIVersion (1.0.5) for the duration when the MSI package has to deviate from others. When we take a fix to the hostfxr.dll itself in future then we can revert to both MSI and other packages to contain the same version (1.0.6 or later) of the package for hostfxr.
- The MSI build related variables have been renamed to correctly reflect the internal MSI version and the MSI file version. 
- For publishing we will continue to publish the packages to Azure blob to the same base version (1.0.1) folder.